### ISSUE TITLE
Add option to change name of subtitles file

### ISSUE THREAD

**Issue by Mrtenz** [McCloudS/subgen#142]
I'm having an issue similar to #131 where Plex doesn't detect the generated subtitles. It seems to be fixed by renaming it to just `.en.srt`. In addition, I'm using Bazarr but _not_ with Subgen directly, as I mainly use the batch endpoint for generating subtitles (yes, weird set up, but I only need to generate subtitles for certain shows, which is not easy with Bazarr directly). Bazarr doesn't detect the `.subgen.large-v3.en.srt` files either, but does detect `.en.srt` files.

The problem with renaming the files manually is that the next time I run Subgen, it wouldn't detect the subtitles and generate them again. It would be nice if there was an option to change the name of the file to be just `.[language].srt`.

**Comment by McCloudS** [comment]
The issue you linked provides a solution to fix plex not detecting them:   https://github.com/McCloudS/subgen/issues/131#issuecomment-2442043532

I can think on a way to add comfortable renaming.  The simplest way right now is manually editing https://github.com/McCloudS/subgen/blob/b1e2bb2fdf41af14861612965ccd545acd55d470/subgen.py#L83

**Comment by McCloudS** [comment]
My Jellyfin shows the files just fine.  The file name being: `10 Things I
Hate About You - S01E02 - I Want You to Want Me.subgen.medium.aa.srt`, `10
Things I Hate About You - S01E02 - I Want You to Want
Me.subgen.medium.zh.srt`, and `10 Things I Hate About You - S01E02 - I Want
You to Want Me.subgen.medium.en.srt`.

Whatever you use in namesublang will be what language it displays as.  I
use Afar (aa) so it doesn't overwrite existing english subtitles or fight
bazarr, in your case, you can probably just set it to 'en' and be fine.
What is yours actually set to?

[image: image.png]

If I have time, I can add a custom subname env variable that will allow you
to customize the naming a bit better.

**Comment by McCloudS** [comment]
My Jellyfin shows the files just fine.  The file name being: `10 Things I Hate About You - S01E02 - I Want You to Want Me.subgen.medium.aa.srt`, `10 Things I Hate About You - S01E02 - I Want You to Want Me.subgen.medium.zh.srt`, and `10 Things I Hate About You - S01E02 - I Want You to Want Me.subgen.medium.en.srt`

Whatever you use in namesublang will be what language it displays as.  I use Afar (aa) so it doesn't overwrite existing english subtitles or fight bazarr, in your case, you can probably just set it to 'en' and be fine. What is yours actually set to?

![Image](https://github.com/user-attachments/assets/79b63058-991e-4b21-b5cc-e95262a8885e)

If I have time, I can add a custom subname env variable that will allow you to customize the naming a bit better.

**Comment by McCloudS** [comment]
Makes sense, the line I pasted actually doesn't exist anymore and has been rewritten a bit, but it still does the same thing.

https://github.com/McCloudS/subgen/blob/49d85bf939e5294cce4696e8b740018e419b077f/subgen.py#L793

If you have `transcribe_or_translate=translate` (and namesublang is empty) the language code will always be 'en' because it's translating to english.  If it's transcribe (and force_language is empty), it will change the language code to the detected language from Whisper.

**Comment by McCloudS** [comment]
`NAMESUBLANG` overrrides the language naming, that needs to be removed/blank.  

Just added `SHOW_IN_SUBNAME_SUBGEN` and `SHOW_IN_SUBNAME_MODEL` which should be getting closer to what you want.

**Comment by McCloudS** [comment]
If you can then, I would set `SHOW_IN_SUBNAME_MODEL` to False and leave `SHOW_IN_SUBNAME_SUBGEN` True, then you'll get a subgen indicator on your file and still get the en (and Bazarr won't muck with it).  The other option, is use Subgen as the Bazarr Whisper provider.

So then subgen would output "video.subgen.en.srt" and Bazarr would be outputting "video.en.srt".

**Comment by McCloudS** [comment]
Multi-stream support in subgen is pretty iffy.  There isn't support to generate multiple for a different audio streams at this time.  Your best bet it using Bazarr to handle that with a language profile, or you can try toying with `PREFERRED_AUDIO_LANGUAGES` and passing it the language stream you want transcribed/translated.
